common/errors.go

```diff
@@ -1973,7 +2024,17 @@ func IsRetryableTowardsUpstream(err error) bool {
 		// Request too-large -> No Retry
 		ErrCodeEndpointUnauthorized,
 		ErrCodeEndpointRequestTooLarge,
-	)
+	) {
+		return false
+	}
+
+	// If the upstream is hitting capacity limits -> no retry
+	if IsCapacityIssue(err) {
+		return false
+	}
```

Comment:
Is this a breaking change compared to current behavior?

Reply:
we have this referenced in two places in the current code and encapsulating it inside the `isRetryableTowardUpstream` would make our code cleaner

here since we are returning `not retryable` for `IsCapacityIssue `, should cover the case here

https://github.com/erpc/erpc/blob/671178233890f88c0d1918d2b209eeec2bba214b/upstream/failsafe.go#L312

and same logic here

https://github.com/erpc/erpc/blob/671178233890f88c0d1918d2b209eeec2bba214b/erpc/networks.go#L279

Reply:
Makes sense, I'm wondering why we didn't have it inside before. My guess is because we didn't have network-level retryable logic, am I correct?

Reply:
Ah nice find!